pkg/helpers/resourcerequirement/resourcerequirement.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Contributors to the Open Cluster Management project
+
+package resourcerequirement
+
+import (
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	operatorv1 "open-cluster-management.io/api/operator/v1"
+)
+
+// ResourceRequirement is for templating resource requirement
+type ResourceRequirement struct {
+	Type                 string
+	ResourceRequirements []byte
+}
+
+func NewResourceRequirement(resourceType string, limits, requests map[string]string) (*ResourceRequirement, error) {
+	if len(limits)+len(requests) == 0 {
+		if resourceType == string(operatorv1.ResourceQosClassResourceRequirement) {
+			return nil, fmt.Errorf("resource type is %s but both limits and requests are not set", resourceType)
+		}
+		return &ResourceRequirement{
+			Type: resourceType,
+		}, nil
+	}
+	resourceType = string(operatorv1.ResourceQosClassResourceRequirement)
```

Comment:
we need to return error if the ResourceQosClassResourceRequirement is not correctly set.